src/Components/Blazor/Build/test/RenderingRazorIntegrationTest.cs

```diff
@@ -361,26 +362,27 @@ public void SupportsTwoWayBindingForTextboxes()
                 }");
             var myValueProperty = component.GetType().GetProperty("MyValue");
 
+            var renderer = new TestRenderer();
+
             // Assert
-            var frames = GetRenderTree(component);
+            Action<UIEventArgs> setter = null;
+            var frames = GetRenderTree(renderer, component);
             Assert.Collection(frames,
                 frame => AssertFrame.Element(frame, "input", 3, 0),
                 frame => AssertFrame.Attribute(frame, "value", "Initial value", 1),
                 frame =>
                 {
                     AssertFrame.Attribute(frame, "onchange", 2);
-
-                    // Trigger the change event to show it updates the property
-                    ((Action<UIEventArgs>)frame.AttributeValue)(new UIChangeEventArgs
-                    {
-                        Value = "Modified value"
-                    });
-                    Assert.Equal("Modified value", myValueProperty.GetValue(component));
+                    setter = Assert.IsType<Action<UIEventArgs>>(frame.AttributeValue);
                 });
+
+            // Trigger the change event to show it updates the property
+            await renderer.Invoke(() => setter(new UIChangeEventArgs { Value = "Modified value", }));
+            Assert.Equal("Modified value", myValueProperty.GetValue(component));
```

Comment:
Now that I've *improved* some of the bind functionality with polyfills these tests require dispatching to the sync context.

Reply:
But events are still processed synchronously, aren't they? They absolutely have to be in the WebAssembly case, otherwise we'll kill certain APIs that are only allowed synchronously.

Is is still possible to have unit tests that show event processing happens synchronously?

Reply:
Is it OK because in the WebAssembly case, dispatched calls always get executed synchronously?

Reply:
Yes, everything still happens synchronously unless you override `OnHandleEventAsync` and do something async there before calling the delegate.